src/main/java/hu/elte/eserial/builder/MapBuilder.java

```diff
@@ -0,0 +1,110 @@
+package hu.elte.eserial.builder;
+
+import hu.elte.eserial.exception.EserialBuilderMismatchException;
+import hu.elte.eserial.exception.EserialInstantiationException;
+import hu.elte.eserial.exception.EserialInvalidMethodException;
+import hu.elte.eserial.util.TypeUtils;
+
+import java.lang.reflect.ParameterizedType;
+import java.lang.reflect.Type;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.ConcurrentNavigableMap;
+import java.util.concurrent.ConcurrentSkipListMap;
+
+/**
+ * Builds Map objects.
+ */
+public class MapBuilder extends AbstractBuilder{
+
+    /**
+     * Constructs a {@link MapBuilder} and sets the {@code mapType} in it.
+     *
+     * @param mapType {@link Map} class to be used in the {@link AbstractBuilder#build} method
+     */
+    MapBuilder(Type mapType) {
+        super(mapType);
+    }
+
+    /**
+     * @param initializationObject {@inheritDoc}
+     * @param <T> {@inheritDoc}
+     * @return a map of the given class and initialized with the map parameter
+     */
+    @Override
+    public <T> T build(Object initializationObject) {
```

Comment:
# This build method expects a Map<Object, Object> and builds the corresponding objects for the key and the value of each entry. However this is not what we agreed to do. If you look at the MapSerializer class, you'll see that our method for serializing Maps is to convert it to a list of entries, each entry containing a key and a value field.
# Let's see this map for example:

```
Map<String, String> map = new HashMap<>();
map.put("a", "A");
map.put("b", "B");
```
# The JSON output for this will be the following:
```
[
  {
    "key": "a",
    "value": "A"
  },
  {
    "key": "b",
    "value": "B"
  }
]
```
# This is not what this build method expects. It expects this JSON format:
```
{
  "a": "A",
  "b": "B"
}
```
# which is totally fine by the way and as it's a valid JSON. It's just not what we agreed to do. Also in your implementation it would also be okay to have a compound key in a map. But that's not valid in JSON. That's why we decided to only use the "list of key-value objects" strategy.

# If we would like to have separate Map serializing/deserializing for simple and compound objects, I'm fine with that, but that means that this build method should be extended to handle the "list of key-value objects" strategy AND the current one, too and that the MapMapper and MapSerializer classes have to be updated. @rolandsz, what do you think?

-------------------------------------------------

# I don't think it's a good idea to support two separate procedures for map serialization and deserialization because it adds unnecessary complexity (even though it might reduce the size of the output) . The current serialization workflow of maps is generalized - why would we add a specialized version as well? It can be an optimization in the future but I think it's out of scope of this assignment as for now.